runtime/common/EvolveResult.h

```diff
@@ -100,20 +100,40 @@ class evolve_result {
     return final_state;
   }
 
-  std::optional<std::vector<state>> get_intermediate_states() {
-    return intermediate_states;
+  std::vector<state> get_intermediate_states() {
+    if (intermediate_states.has_value()) {
+      return intermediate_states.value();
+    } else {
+      throw std::runtime_error(
+          "no intermediate states have been computed in the call to evolve");
+    }
   }
 
-  std::optional<std::vector<observe_result>> get_final_expectation_values() {
-    return final_expectation_values;
+  std::vector<observe_result> get_final_expectation_values() {
+    if (final_expectation_values.has_value()) {
+      return final_expectation_values.value();
+    } else {
```

Comment:
else after return. this might go away though since the spec is moving toward simple struct with public std::optional values.